aws/resource_aws_storagegateway_fsx_associate_file_system_test.go

```diff
@@ -0,0 +1,434 @@
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsStorageGatewayFsxAssociateFileSystemDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsStorageGatewayFsxAssociateFileSystemConfig_Required(rName, domainName, username),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsStorageGatewayFsxAssociateFileSystemExists(resourceName, &fsxFileSystemAssociation),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsStorageGatewayFsxAssociateFileSystem(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSStorageGatewayFsxAssociateFileSystem_disappears_storageGateway(t *testing.T) {
+	var fsxFileSystemAssociation storagegateway.FileSystemAssociationInfo
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_storagegateway_fsx_associate_file_system.test"
+	domainName := testAccRandomDomainName()
+	username := "Admin"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(storagegateway.EndpointsID, t) },
+		ErrorCheck:   testAccErrorCheck(t, storagegateway.EndpointsID),
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsStorageGatewayFsxAssociateFileSystemDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsStorageGatewayFsxAssociateFileSystemConfig_Required(rName, domainName, username),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsStorageGatewayFsxAssociateFileSystemExists(resourceName, &fsxFileSystemAssociation),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsStorageGatewayGateway(), "aws_storagegateway_gateway.test"),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+//
+// func TestAccAWSStorageGatewayFsxAssociateFileSystem_disappears_fsxFileSystem(t *testing.T) {
+// 	var fsxFileSystemAssociation storagegateway.FileSystemAssociationInfo
+// 	rName := acctest.RandomWithPrefix("tf-acc-test")
+// 	resourceName := "aws_storagegateway_fsx_associate_file_system.test"
+// 	username := "Admin"
+
+// 	resource.ParallelTest(t, resource.TestCase{
+// 		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(storagegateway.EndpointsID, t) },
+// 		ErrorCheck:   testAccErrorCheck(t, storagegateway.EndpointsID),
+// 		Providers:    testAccProviders,
+// 		CheckDestroy: testAccCheckAwsStorageGatewayFsxAssociateFileSystemDestroy,
+// 		Steps: []resource.TestStep{
+// 			{
+// 				Config: testAccAwsStorageGatewayFsxAssociateFileSystemConfig_Required(rName, username),
+// 				Check: resource.ComposeTestCheckFunc(
+// 					testAccCheckAwsStorageGatewayFsxAssociateFileSystemExists(resourceName, &fsxFileSystemAssociation),
+// 					testAccCheckResourceDisappears(testAccProvider, resourceAwsFsxWindowsFileSystem(), "aws_fsx_windows_file_system.test"),
+// 				),
+// 				ExpectNonEmptyPlan: true,
+// 			},
+// 		},
+// 	})
+// }
```

Comment:
Instead of commenting this out, add a `t.Skip()` that explains why it is being skipped. It would also be useful to open a support case, since it shouldn't hang in the `DELETING` state like that.

Reply:
yes this is definitely a bug. I reported a bug internally last week. My intent is to do a follow-on PR when the issue is fixed. Unfortunately the issue tracker is not exposed externally. if theres something else i can do to show clarity for yall, please let me know.